dvc/repo/__init__.py

```diff
@@ -239,7 +239,14 @@ def __str__(self):
     def index(self):
         from dvc.repo.index import Index
 
-        return Index(self)
+        return Index.from_repo(self)
+
+    def ensure_graph_correctness_with(self, stages=None, callback=None):
+        if not getattr(self, "_skip_graph_checks", False):
+            new = self.index.update(stages)
+            if callable(callback):
+                callback()
```

Comment:
When `self.index` is called, it might go on and collect stages. So the `callback` is there to signify that collection was completed, and that we are moving to check graph.

This is used in `dvc add` to show `Checking graph` status output.